### DISCUSSION TITLE
Feature Request | Question: Reference the input value in the error message?

### DISCUSSION THREAD

**Discussion by brianjlacy** [colinhacks/zod#3121] [Q&A] [answered]
**On validation, I would like to be able to reference the input value in the error message.**

For example, given that my schema looks like this...
```
const mySchema = z.object({
  numberOfDays: z.number({
    invalid_type_error: "Sorry, that is not a valid number."
});
```

...then I would like to _instead_ be able to return an error message like:
**"Sorry, 'seven' is not a valid number."**

I'm imagining something like either:

```invalid_type_error: "Sorry, '{{value}}' is not a valid number."```

Or better yet:

```invalid_type_error: (val) => `Sorry, '${val}' is not a valid number."```

Is this already possible? If not, could this be added in the future?

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const mySchema = z.object( {
    numberOfDays: z.number( {
        errorMap: ( issue, ctx ) => {
            if ( issue.code == 'invalid_type' )
                return { message: `Sorry, '${ ctx.data }' is not a valid number.` }
            return { message: ctx.defaultError }
        }
    } )
} )
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
